internal/sdkprovider/service/kafka/kafka_native_acl.go

```diff
@@ -0,0 +1,169 @@
+	},
+	"operation": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.StringInSlice(kafka.OperationTypeChoices(), false),
+		Description:  userconfig.Desc("The operation").ForceNew().PossibleValuesString(kafka.OperationTypeChoices()...).Build(),
+	},
+	"permission_type": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.StringInSlice(kafka.KafkaAclPermissionTypeChoices(), false),
+		Description:  userconfig.Desc("The permission type").ForceNew().PossibleValuesString(kafka.KafkaAclPermissionTypeChoices()...).Build(),
+	},
+}
+
+func resourceKafkaNativeACLCreate(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	var req kafka.ServiceKafkaNativeAclAddIn
+	err := schemautil.ResourceDataGet(d, &req)
+	if err != nil {
+		return err
+	}
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+
+	acl, err := client.ServiceKafkaNativeAclAdd(
+		ctx,
+		project,
+		serviceName,
+		&req,
+	)
+	if err != nil {
+		return err
+	}
+
+	err = schemautil.ResourceDataSet(aivenKafkaNativeACLSchema, d, acl)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(schemautil.BuildResourceID(project, serviceName, acl.Id))
+	return resourceKafkaNativeACLRead(ctx, d, client)
+}
+
+func resourceKafkaNativeACLRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	project, serviceName, aclID, err := schemautil.SplitResourceID3(d.Id())
+	if err != nil {
+		return err
+	}
+
+	acl, err := client.ServiceKafkaNativeAclGet(
+		ctx,
+		project,
+		serviceName,
+		aclID,
+	)
+	if err != nil {
+		return err
+	}
+
+	if acl == nil {
+		return fmt.Errorf(
+			errmsg.AivenResourceNotFound,
+			"aiven_kafka_native_acl",
+			aclID,
+		)
+	}
+
+	err = schemautil.ResourceDataSet(aivenKafkaNativeACLSchema, d, acl)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceKafkaNativeACLDelete(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	project, serviceName, aclID, err := schemautil.SplitResourceID3(d.Id())
+	if err != nil {
+		return err
+	}
+
+	_, err = client.ServiceKafkaNativeAclDelete(
+		ctx,
+		project,
+		serviceName,
+		aclID,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ResourceKafkaNativeACL() *schema.Resource {
+	return &schema.Resource{
+		Description:   userconfig.Desc(`Manages native acls in [kafka service](https://aiven.io/docs/products/kafka/concepts/acl)`).Build(),
+		CreateContext: common.WithGenClient(resourceKafkaNativeACLCreate),
+		ReadContext:   common.WithGenClient(resourceKafkaNativeACLRead),
+		DeleteContext: common.WithGenClient(resourceKafkaNativeACLDelete),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: schemautil.DefaultResourceTimeouts(),
+		Schema:   aivenKafkaNativeACLSchema,
+	}
+}
```

Comment:
nit: let's move this under the schema, that's a common place to search for this block )